Optical radiation review and calculations

In [1]:
from sympy import *

When dealing with blackbody radiation, the spectral radiance can be expressed with this frequently encountered form of the Planck equation:

$$\large L_{\lambda} = \frac{c_{1}}{\pi n^2 \lambda^5} \frac{1}{e^\frac{c_{2}}{n\lambda T}-1} $$

which has units of $\large \frac{W}{m^2 sr-m}$

In [2]:
k = 1.38*10**-23       #Boltzmann's constant [J/K]
c = 299_792_458        #Exact speed of light [m/s]
h = 6.626*10**-34      #Planck's constant [Js]
c1 = 2*pi*h*c**2       #first radiation constant [Wm^2]
c2 = (h*c)/k           #second radiation constant [m-K]
n = 1.0028             #index of refraction for air

λ = 10*10**-6          #user specified wavelength [m] 
T = 300                #user specified temperature [K]


In [3]:
L = ((c1/((pi*n**2)*λ**5)*(1/(exp(c2/(n*λ*T))-1))))*10**-6
L.n()

9.98059810818411

In the cell above we multiplied the result by $10^{-6}$ to convert to  $\large \frac{W}{m^2 sr-\mu m}$

Since blackbody radiation is Lambertian, if spectral radiant exitance is needed, use the following:

$$\large M_{\lambda} = \pi L_{\lambda}$$

In [4]:
M = pi*L.n()
M.n()

31.3549736951034

$\large Application:$

You're out in a field at night and spot a glowing orb that's stationary and is located roughly 3 meters from your location. For some reason, instead of taking a video of it, you mount a power meter to a tripod and detect a reading of 10$\mu W$. The detector has an active area of 3.14$mm^2$. The orb is on-axis to the detector but your detector is tilted by 30 degrees. You estimate the orb to have a diameter of 5cm. 

1) What is the total output power from the orb at your location?

2) What is the temperature of the orb?

3) What is the wavelength of maximum spectral radiant exitance?

In [6]:
#define specs
l = 3           #[m] distance from your location to the orb
d = 0.05        #[m] diameter of orb
Φ = 10*10**-6      #[W] radiant power measured at the detector
A = 3.14*10**-6 #[m^2] area of the detector converted to square meters

Treating the orb as a point source, we apply the inverse square law of irradiance:
$$ E = \frac{I \ cos(\theta)}{l^2} $$

solving for radiant intensity we have:
$$ I = \frac{El^2}{cos(\theta)} $$

In [7]:
#solve for irradiance, E, first
E = Φ/A
E    #[W/m^2]

3.184713375796178

In [8]:
#solve for radiant intensity
I = (E*l**2)/cos(30)
I.n()    #[W/sr]

185.816213740968

The orb is a sphere. Therefore in order to obtain total radiant output power at this location, we multiply the obtained radiant intensity by the solid angle of the sphere, $4\pi$.

In [9]:
Φ = I*4*pi
Φ.n()   #total radiant power [W] at your location

2335.03540802599

Next, to determine the temperature of the orb we must first determine the orb's radiance. We have established the orb is a blackbody radiator and thus Lambertian, therefore we can use the Lambertian approximation:
$$I = LA_{p}$$

where $A_{p}$ is the area of the source projected in the direction of the detector:
$$A_{p} = A_{s}cos\theta$$
and $$A_{s} = \pi r^2 = \frac{\pi d^2}{4}$$

resulting in

$$L = \frac{4I}{\pi d^2 cos(\theta)}$$

In [10]:
#solving for radiance we have:
L = (4*I.n())/(pi*d**2 *cos(30))
L.n()

613513.977496832

Now to determine the orb's temperature, we apply the Stefan-Boltzmann law which is most often seen in terms of radiant exitance:

$$M = \sigma T^4$$

Using the relationship between radiant exitance and radiance defined previously above we have:

$$L = \frac{\sigma T^4}{\pi} $$
solving for temperature T we have:
$$T = \sqrt[4]{\frac{L\pi}{\sigma}}$$

In [11]:
#solve for temperature
σ = 5.670*10**-8  #[W/m^2-K^5-um] Stefan-Boltzmann constant
T = real_root((L*pi)/σ, 4)
T.n()

2414.61458093719

Lastly, to determine the wavelength of max spectral radiant exitance, we use the Wien displacement law:
$$\lambda_{max}T = 2897\ \mu m-K$$

In [12]:
λ = 2897/T
λ.n()    #[μm]

1.19977739837700

In summary, the orb had a total output power of $2,335.03 W$, a temperature of $2,414.61K \ (2,141.46\ C\ or\ 3,886.628\ F)$ and a wavelength of max spectral radiant exitance of $1.20\mu m$.